internal/cli/generate_tiltfile.go

```diff
@@ -0,0 +1,175 @@
+// the user is running in an interactive (TTY) session.
+//
+// The generateTiltfileResult value is ALWAYS set, even on error.
+func maybeGenerateTiltfile(tfPath string) (generateTiltfileResult, error) {
+	if !term.IsTerminal(int(os.Stdin.Fd())) || !term.IsTerminal(int(os.Stdout.Fd())) {
+		return generateTiltfileNonInteractive, nil
+	}
+
+	if absPath, err := filepath.Abs(tfPath); err != nil {
+		// the absolute path is really just to improve CLI output, if the path
+		// itself is so invalid this fails, we'll catch + report it via the
+		// logic to determine if a Tiltfile already exists
+		tfPath = absPath
+	}
+
+	if hasTiltfile, err := checkTiltfileExists(tfPath); err != nil {
+		// either Tiltfile path is totally invalid or there's something like
+		// a permissions error, so report it & exit
+		return generateTiltfileError, err
+	} else if hasTiltfile {
+		// Tiltfile exists, so don't prompt to generate one
+		return generateTiltfileAlreadyExists, nil
+	}
+
+	t, restoreTerm, err := setupTerm(os.Stdin)
+	if err != nil {
+		return generateTiltfileError, nil
+	}
+
+	lineCount := 0
+	var postFinishMessage string
+	defer func() {
+		_ = restoreTerm()
+
+		if postFinishMessage != "" {
+			// NOTE: pre-win10, there's no support for ANSI escape codes, and
+			// 	it's not worth the headache to deal with Windows console API
+			// 	for this, so the output isn't cleared there
+			if err == nil && runtime.GOOS != "windows" {
+				// erase our output once done on success
+				// \033[%d -> move cursor up %d rows
+				// \r      -> move cursor to first column
+				// \033[J  -> clear output from cursor to end of stream
+				fmt.Printf("\033[%dA\r\033[J", lineCount)
+			}
+
+			fmt.Println(postFinishMessage)
+		}
+	}()
+
+	// Offer to create a Tiltfile
+	var intro bytes.Buffer
+	intro.WriteString(tiltfileDoesNotExistWarning(tfPath))
+	intro.WriteString(`
+Tilt can create a sample Tiltfile for you, which includes
+useful snippets to modify and extend with build and deploy
+steps for your microservices.
+`)
+	intro.WriteString("\n")
+	_, err = t.Write(intro.Bytes())
+	if err != nil {
+		return generateTiltfileError, err
+	}
+	// we track # of lines written to clear the output when done
+	lineCount += bytes.Count(intro.Bytes(), []byte("\n"))
+
+	for {
+		line, err := t.ReadLine()
+		lineCount++
+		if err != nil {
+			// perform a carriage return to ensure we're back at the beginning
+			// of a new line (if user hit Ctrl-C/Ctrl-D, this is necessary; for
+			// any other errors, better to be safe than leave terminal in a bad
+			// state)
+			fmt.Println("\r")
+			if err == io.EOF {
+				// since we have the terminal in raw mode, no signal will be fired
+				// on Ctrl-C, so we manually propagate it here (sending ourselves
+				// a SIGINT signal is not practical)
+				return generateTiltfileUserExit, userExitError
+			}
+			return generateTiltfileError, err
+		}
+
+		line = strings.ToLower(strings.TrimSpace(line))
+		if line == "y" || line == "yes" {
```

Comment:
Up to you if it's worth just checking for 'Y'. Or can we read a single character instead of a line and use that?
```suggestion
		if strings.HasPrefix(line, "y") {
```